src/control.rs

```diff
@@ -1,25 +1,72 @@
+use crate::event_loop::EDT;
+use crate::event_loop::Msg;
 use crate::led::Led;
 use crate::pin::Pin;
+use crate::rgb::{BLUE, GREEN, RED, Rgb};
 
 /// Control logic evaluates button states and changes the light intensity
 pub struct LightControl<'a, P: Pin> {
     pub plus_pin: P,
     pub minus_pin: P,
     pub led: &'a dyn Led,
+    pub rgb: &'a dyn Rgb,
+    pub edt: &'a EDT,
 }
 
-impl <'a, P: Pin> LightControl<'a, P> {
-    pub fn tick(&self) {
+const MSG_BLINK: i32 = 1;
+const MSG_CHECK_BUTTONS: i32 = 2;
+
+const DELAY_CHECK_BUTTONS: u32 = 75;
+const DELAY_BLINK: u32 = 250;
+
+impl<'a, P: Pin> LightControl<'a, P> {
+    pub fn process_message(&self, msg: Msg) {
+        match msg.what {
+            MSG_CHECK_BUTTONS => self.check_buttons(),
+            MSG_BLINK => self.blink_led(msg),
+            _ => {}
+        }
+    }
+
+    fn blink_led(&self, msg: Msg) {
+        if msg.arg1 > 0 {

```

Comment:
I think it would be better to 
- [ ] check arg1 only for the repetition
- [ ] "extract explaining variable" would be nice here

Reply:
Also for this block: I'd use enums for such values instead of const integers.

Reply:
And what if we want arg1 be 0? Maybe try to use core::Option (https://doc.rust-lang.org/core/option/index.html)?